src/AssetControlExtension.php

```diff
@@ -268,6 +269,32 @@ protected function deleteAll($assets)
         }
     }
 
+    /**
+     * Move all assets in the list to the public store
+     *
+     * @param array $assets
+     */
+    protected function swapAll($assets)
+    {
+        if (empty($assets)) {
+            return;
+        }
+
+        $store = $this->getAssetStore();
+
+        // The `swap` method was introduced in the 1.4 release. It wasn't added to the interface to avoid breaking
+        // custom implementation. If it's not available on our store, we fallback to a publish/protect
+        if (method_exists($store, 'swapPublish')) {
```

Comment:
```suggestion
        if (method_exists($store, 'swapPublishAll')) {
```

Reply:
`swapPublish` on `AssetStore` only publishes one file at the time. So it doesn't make sense to call it `swapPublishAll`.

The swap publish method on AssetControllerExtension operates on a list of filename/hash combination.